src/dotnet/commands/dotnet-restore/RestoreCommandParser.cs

```diff
@@ -31,7 +31,28 @@ private static Option[] FullRestoreOptions()
                         "--interactive",
                         LocalizableStrings.CmdInteractiveRestoreOptionDescription,
                         Accept.NoArguments()
-                              .ForwardAs("-property:NuGetInteractive=true")) }).ToArray();
+                            .ForwardAs("-property:NuGetInteractive=true")),
+                    Create.Option(
+                        "--use-lock-file",
+                        LocalizableStrings.CmdUseLockFileOptionDescription,
+                        Accept.NoArguments()
+                            .ForwardAs("-property:RestorePackagesWithLockFile=true")),
+                    Create.Option(
+                        "--locked-mode",
```

Comment:
What is the difference between --locked-mode and --use-lock-file. 

I suspect the naming isn't quite right here yet because this is unclear.

Reply:
`--use-lock-file` is to just bootstrap into this feature so once lock file is present it's always used. Lock file is always modified, if required, by restore but `--locked-mode` prevents that. So this flag will be useful for CI/CD scenarios where you don't want restore to modify lock file, hence enable repeatable builds.